src/Report/JunitReporter.php

```diff
@@ -34,6 +34,10 @@ public function getFormat()
      */
     public function generate(ReportSummary $reportSummary)
     {
+        if (!extension_loaded('dom')) {
```

Comment:
early check would be nice indeed, but since this is a public method, we need to check it here nevertheless